opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -42,26 +43,30 @@ def wrapper(
         **kwargs: typing.Dict[typing.Any, typing.Any]
     ) -> typing.Optional[typing.Any]:
         global _RUNTIME_CONTEXT  # pylint: disable=global-statement
-        if _RUNTIME_CONTEXT is None:
-            # FIXME use a better implementation of a configuration manager to avoid having
-            # to get configuration values straight from environment variables
-            if version_info < (3, 5):
-                # contextvars are not supported in 3.4, use thread-local storage
-                default_context = "threadlocal_context"
-            else:
-                default_context = "contextvars_context"
-
-            configured_context = environ.get(
-                "OPENTELEMETRY_CONTEXT", default_context
-            )  # type: str
-            try:
-                _RUNTIME_CONTEXT = next(
-                    iter_entry_points(
-                        "opentelemetry_context", configured_context
+
+        with _RUNTIME_CONTEXT_LOCK:
+            if _RUNTIME_CONTEXT is None:
+                # FIXME use a better implementation of a configuration manager to avoid having
+                # to get configuration values straight from environment variables
+                if version_info < (3, 5):
+                    # contextvars are not supported in 3.4, use thread-local storage
+                    default_context = "threadlocal_context"
```

Comment:
Is it also necessary to use `_RUNTIME_CONTEXT_LOCK` with `threadlocal_context`?

Reply:
I suspect the problem would exist with `ThreadLocalRuntimeContext` as well, the test was instantiating the runtime context and setting it before the tests though, so it would have masked the underlying problem of both threads trying to set `_RUNTIME_CONTEXT`

Reply:
It doesn't look like it matters which context gets used, just that we have to hold the lock until `_RUNTIME_CONTEXT` is set. This looks like a good approach since we only expect to reach this point once per app anyway.